src/unix/linux.c

```diff
@@ -2305,49 +2305,73 @@ uint64_t uv_get_available_memory(void) {
 
 
 static int uv__get_cgroupv2_constrained_cpu(const char* cgroup,
-                                            uv__cpu_constraint* constraint) {
-  char path[256];
-  char buf[1024];
-  unsigned int weight;
-  int cgroup_size;
+                                            long long* quota) {
+  static const char cgroup_mount[] = "/sys/fs/cgroup";
   const char* cgroup_trimmed;
+  char buf[1024];
+  char full_path[256];
+  char path[256];
   char quota_buf[16];
+  char* last_slash;
+  int cgroup_size;
+  long long limit;
+  long long min_quota;
+  long long period;
 
   if (strncmp(cgroup, "0::/", 4) != 0)
     return UV_EINVAL;
 
   /* Trim ending \n by replacing it with a 0 */
   cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
-  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first slash */
+  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first \n */
+  min_quota = LLONG_MAX;
 
-  /* Construct the path to the cpu.max file */
-  snprintf(path, sizeof(path), "/sys/fs/cgroup/%.*s/cpu.max", cgroup_size,
-           cgroup_trimmed);
+  /* Construct the path to the cpu.max files */
+  snprintf(path, sizeof(path), "%s/%.*s/cgroup.controllers", cgroup_mount,
+           cgroup_size, cgroup_trimmed);
 
-  /* Read cpu.max */
+  /* Read controllers, if not exists, not really a cgroup */
   if (uv__slurp(path, buf, sizeof(buf)) < 0)
     return UV_EIO;
 
-  if (sscanf(buf, "%15s %llu", quota_buf, &constraint->period_length) != 2)
-    return UV_EINVAL;
+  snprintf(path, sizeof(path), "%s/%.*s", cgroup_mount, cgroup_size,
+           cgroup_trimmed);
 
-  if (strncmp(quota_buf, "max", 3) == 0)
-    constraint->quota_per_period = LLONG_MAX;
-  else if (sscanf(quota_buf, "%lld", &constraint->quota_per_period) != 1)
-    return UV_EINVAL; // conversion failed
+  /* Read cpu.max in the whole file-hierarchy */
+  while(strncmp(path, cgroup_mount, strlen(cgroup_mount)) == 0) {
```

Comment:
```suggestion
  while (strncmp(path, cgroup_mount, strlen(cgroup_mount)) == 0) {
```
I don't have a cgroupsv2 system at hand to check but can you have a file that looks like this?
```
/sys/fs/cgroup/foo
/sys/fs/cgroup/bar
/something/completely/different
/sys/fs/cgroup/baz
```

Reply:
That's not gonna happen. The `cgroup` argument will be the one giving us the file-hierarchy. Consider this simple diff. Also keep in mind that libuv double check the cgroup argument format, and also it is optained from reading `/proc/self/cgroup` (see `uv__get_constrained_cpu`)

```diff
diff --git a/src/unix/linux.c b/src/unix/linux.c
index 73431ec0..f9e43e39 100644
--- a/src/unix/linux.c
+++ b/src/unix/linux.c
@@ -2321,6 +2321,8 @@ static int uv__get_cgroupv2_constrained_cpu(const char* cgroup,
   if (strncmp(cgroup, "0::/", 4) != 0)
     return UV_EINVAL;

+  fprintf(stderr, "cgroup: %s\n", cgroup);
+
   /* Trim ending \n by replacing it with a 0 */
   cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
   cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first \n */
@@ -2340,6 +2342,7 @@ static int uv__get_cgroupv2_constrained_cpu(const char* cgroup,
   /* Read cpu.max in the whole file-hierarchy */
   while(strncmp(path, cgroup_mount, strlen(cgroup_mount)) == 0) {
     snprintf(full_path, sizeof(full_path), "%s/cpu.max", path);
+    fprintf(stderr, "full_path: %s\n", full_path);

     /* Silently ignore and continue if the file does not exist */
     if (uv__slurp(full_path, quota_buf, sizeof(quota_buf)) < 0)
```

This is the output:
```
cgroup: 0::/user.slice/user-1000.slice/user@1000.service/app.slice/tmux-spawn-0a4efa52-dc83-4826-bf02-77770535152b.scope

full_path: /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/app.slice/tmux-spawn-0a4efa52-dc83-4826-bf02-77770535152b.scope/cpu.max
full_path: /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/app.slice/cpu.max
full_path: /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/cpu.max
full_path: /sys/fs/cgroup/user.slice/user-1000.slice/cpu.max
full_path: /sys/fs/cgroup/user.slice/cpu.max
full_path: /sys/fs/cgroup/cpu.max
uv_available_parallelism: 16
```